#!/usr/bin/env python
PACKAGE = "image_stitcher"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("stitcher_num_frames", int_t,    0, "number of frames to stitch", 20, 0, 100)
gen.add("registr_resol",       double_t, 0, "registr_resol",              0.6, -1, 1)
gen.add("seam_est_resol",      double_t, 0, "seam_est_resol",             0.1, -1, 1)
gen.add("compose_resol",       double_t, 0, "compose_resol",              -1, -1, 1)
gen.add("conf_thresh",         double_t, 0, "conf_thresh",                1, -1, 1)
gen.add("do_wave_correct",     bool_t,   0, "do_wave_correct",            True)

exit(gen.generate(PACKAGE, "stitcher", "Stitcher"))
